Lib/urllib/parse.py

```diff
@@ -734,6 +734,10 @@ def unquote_plus(string, encoding='utf-8', errors='replace'):
 
     unquote_plus('%7e/abc+def') -> '~/abc def'
     """
+    if isinstance(string, bytes):
```

Comment:
according to the documentation https://docs.python.org/3/library/urllib.parse.html?highlight=urllib#urllib.parse.unquote_plus `string` must be a `str` type.

If you need parse bytes you must use `unquote_to_bytes`. IMO here you need prove if 
string is a str

```
if !isinstance(string, str):
    raise TypeError ('a str object is required')
```

Reply:
I think handling both bytes and str would make it more like unquote in 3.9 (https://docs.python.org/3.9/library/urllib.parse.html?highlight=urllib#urllib.parse.unquote).

I would expect them to behave the same and accept the same types.